### ISSUE TITLE
A hang occurs after the trainval_net.py finishing its first epoch’s training

### ISSUE THREAD

**Issue by neulrl** [jwyang/faster-rcnn.pytorch#150]
- OS: Ubuntu 16.04
- PyTorch version: pytorch0.3.1
- Python version: python3.5(anaconda3)
- CUDA version: 8.0
- GPU models : Titanxp
- dataset； voc2007
A hang occurs after the trainval_net.py finishing its first epoch’s training
![image](https://user-images.githubusercontent.com/29855057/39407443-be52a214-4bf8-11e8-8781-4f463732296a.png)
I found that the process was in a state of sleep.
![default](https://user-images.githubusercontent.com/29855057/39407475-53bc0ec6-4bf9-11e8-8cd3-44c567e86a9a.png)
After killing the process with CTRL+C:
![default](https://user-images.githubusercontent.com/29855057/39407486-6c7e3114-4bf9-11e8-8718-663d55670226.png)
I use ipdb to set breakpoint on line 314 (line 312 in original program) and single step debug the program.
It also got stuck at line 314：
![image](https://user-images.githubusercontent.com/29855057/39407572-70f71da4-4bfa-11e8-8e6e-b3c5925b2683.png)

Does anyone know how to solve the problem?

**Comment by neulrl** [comment]
I have solved this problem by setting num_workers(one of the arguments of Dataloader) to 0. Due to this page：https://discuss.pytorch.org/t/guidelines-for-assigning-num-workers-to-dataloader/813 , it means that it’s the main process that will do the data loading when needed.
After that I tried to solve this problem without setting num_workers to 0 (to make code run faster)and find an issue at https://github.com/pytorch/pytorch/issues/1355, I had tried these ways to solve my problem:
'''from multiprocessing import set_start_method
set_start_method('spawn')'''

'''
cv2.setNumThreads(0)'''
'''
'''
Setting pin_memory=True(default is Fasle)'''

However, these methods didn’t solve my problem.
